Mathlib/LinearAlgebra/BilinearForm/Properties.lean

```diff
@@ -268,11 +268,9 @@ noncomputable def dualBasis (B : BilinForm K V) (hB : B.Nondegenerate) (b : Basi
 theorem dualBasis_repr_apply
     (B : BilinForm K V) (hB : B.Nondegenerate) (b : Basis ι K V) (x i) :
     (B.dualBasis hB b).repr x i = B x (b i) := by
-  #adaptation_note /-- https://github.com/leanprover/lean4/pull/4814
-  we did not need the `@` in front of `toDual_def` in the `rw`.
-  I'm confused! -/
+  have := FiniteDimensional.of_fintype_basis b
```

Comment:
Why is this change needed now?

Reply:
The hypothesis *could* be found by unification in `toDual`. But elaborating `rw [..., toDual_def]` requires filling in the hypothesis before the unification gets started. It's the same case in `apply_dualBasis_left` below, where we also need to add the same instance to the context.